data_steward/cdr_cleaner/clean_cdr_engine.py

```diff
@@ -140,6 +149,226 @@ def clean_dataset(project=None, statements=None, data_stage=stage.UNSPECIFIED):
+    destination_table = get_destination_ref(project_id, query_dict)
+
+    job_config.destination = destination_table
+    job_config.use_legacy_sql = query_dict.get(cdr_consts.LEGACY_SQL, False)
+    # allow_large_results can only be used if use_legacy_sql=True
+    job_config.allow_large_results = job_config.use_legacy_sql
+    job_config.write_disposition = query_dict.get(cdr_consts.DISPOSITION,
+                                                  bq_consts.WRITE_EMPTY)
+    return job_config
+
+
+def run_queries(client, query_list, rule_info):
+    """
+    Runs queries from the list of query_dicts
+
+    :param client: BigQuery client
+    :param query_list: list of query_dicts generated by a cleaning rule
+    :param rule_info: contains information about the query function
+    :return: integers indicating the number of queries that succeeded and failed
+    """
+    query_count = len(query_list)
+    successes, failures = [], []
+    for index, query_dict in enumerate(query_list):
+        try:
+            LOGGER.info(
+                ce_consts.QUERY_RUN_MESSAGE_TEMPLATE.render(
+                    index, query_count, **rule_info, **query_dict))
+            job_config = generate_job_config(client.project, query_dict)
+
+            module_name = rule_info[cdr_consts.MODULE_NAME]
+            query_job = client.query(query=query_dict.get(cdr_consts.QUERY),
+                                     job_config=job_config,
+                                     job_id_prefix=f'clean_{module_name}_')
+
+            # wait for job to complete
+            query_job.result()
+        except (GoogleCloudError, TOError) as exp:
+            LOGGER.exception(
+                ce_consts.FAILURE_MESSAGE_TEMPLATE.render(client.project,
+                                                          **rule_info,
+                                                          **query_dict,
+                                                          exception=exp))
+            failures.append(query_dict[cdr_consts.QUERY])
+            continue
+        else:
+            successes.append(query_dict[cdr_consts.QUERY])
+    return successes, failures
+
+
+def infer_rule(clazz, project_id, dataset_id, sandbox_dataset_id):
+    """
+    Extract information about the cleaning rule
+
+    :param clazz: Clean rule class or old style clean function
+    :param project_id: identifies the project
+    :param dataset_id: identifies the dataset to clean
+    :param sandbox_dataset_id: identifies the dataset to store sandbox tables
+    :return: rule_info: dictionary of information about the rule
+        keys:
+            query_function: function that generates query_list
+            setup_function: function that sets up the tables for the rule
+            function_name: name of the query function
+            module_name: name of the module containing the function
+            line_no: points to the source line where query_function is
+    """
+    if inspect.isclass(clazz):
```

Comment:
This is good, better to do `inspect.isclass()` than try catching. It seems prudent to also do `issubclass(clazz, BaseCleaningRule)`, is there a reason to exclude that check? I recommend also raising an error if `inspect.isclass(clazz) and not issubclass(clazz, BaseCleaningRule)`.